src/main/java/org/prebid/server/privacy/gdpr/TcfDefinerService.java

```diff
@@ -143,10 +150,17 @@ public TcfDefinerService(GdprConfig gdprConfig,
                         dispatchToService(gdprInfoWithCountry, allowAllTcfResponseCreator, tcf2Strategy, gdprStrategy));
     }
 
-    private boolean isGdprEnabled(AccountGdprConfig accountGdprConfig) {
-        return accountGdprConfig != null && accountGdprConfig.getEnabled() != null
-                ? accountGdprConfig.getEnabled()
-                : gdprEnabled;
+    private boolean isGdprEnabled(AccountGdprConfig accountGdprConfig, MetricName requestType) {
+        if (requestType == null) {
+            return accountGdprConfig != null && accountGdprConfig.getEnabled() != null
+                    ? accountGdprConfig.getEnabled()
+                    : gdprEnabled;
+        }
+        final EnabledForRequestType enabledForRequestType = accountGdprConfig != null
+                ? accountGdprConfig.getEnabledForRequestType()
+                : null;
+        final Boolean enabled = enabledForRequestType != null ? enabledForRequestType.isEnabledFor(requestType) : null;
+        return enabled != null ? enabled : gdprEnabled;
```

Comment:
Probably, it's better to get accountEnabled from retrieved account also.
```
 final Boolean accountGdprEnabled = accountGdprConfig != null && accountGdprConfig.getEnabled() != null;

        if (requestType == null) {
            return ObjectUtils.firstNonNull(accountGdprEnabled, gdprEnabled);
        }

        final EnabledForRequestType enabledForRequestType = accountGdprConfig != null
                ? accountGdprConfig.getEnabledForRequestType()
                : null;
        final Boolean enabledForType = enabledForRequestType != null
                ? enabledForRequestType.isEnabledFor(requestType)
                : null;

        return ObjectUtils.firstNonNull(enabledForType, accountGdprEnabled, gdprEnabled);
```

Reply:
accountGdprConfig.getEnabled() should not have impact on auction typed requests

Reply:
After discussion with Bret we decide that integration_enabled should override account_enabled. So @DGarbar is right, pull request updated. Thanks